docs/api/masking.md

```diff
@@ -0,0 +1,899 @@
+    csvProcessor.maskStream(inputStream, output, factory, dataMaskingOptions,  Collections.emptySet(), 				Collections.emptyMap()); 
+    
+    //print/check output here
+}
+```
+
+### JSON, DICOM, XLS, XLSX, XML, PARQUET, VCF
+
+Similar to CSV case but changing the DataTypeFormat to the factory call as well as the dataset options should be set to `null`. The `InputStream` should be a regular input stream to the contents of the data
+(like a FileInputStream).
+
+
+### JDBC
+
+```java
+
+private static Connection getConnection() throws ClassNotFoundException, SQLException
+{
+        String DBUrl      = "jdbc:postgresql://localhost/test";
+        String DBUsername = "postgres";
+        String DBPassword = ""; // Need password
+
+        //Class.forName(DBDriver);
+        return DriverManager.getConnection(DBUrl, DBUsername, DBPassword);
+ }
+
+public void testDatabase() {
+	String schemaName = "s1";
+	String tableName = "people";
+	
+	ConfigurationManager configurationManager = new ConfigurationManager(new DefaultMaskingConfiguration());
+   MaskingProviderFactory factory = new MaskingProviderFactory(configurationManager, Collections.emptyMap());
+    
+   String path = "name";
+
+   Map<String, DataMaskingTarget> identifiedTypes = new HashMap<>();
+   identifiedTypes.put(path, new DataMaskingTarget(ProviderType.HASH, path));
+            
+	InputStream jdbcStream  = new ResultSetInputStream(getConnection(), schemaName, tableName);
+	
+	FormatProcessor formatProcessor = FormatProcessorFactory.getProcessor(DataTypeFormat.JDBC);
+	
+	PrintStream output = new JDBCOutputStream(getConnection(), "public", targetTable,
+                true, "public", "test", true);
+   
+   formatProcessor.maskStream(jdbcStream, output, factory, dataMaskingOptions, Collections.emptySet(),  Collections.emptyMap());
+	
+}
+```
+
+## Path notations for various types
+
+### CSV
+
+!INCLUDE "../common/csv-path-notation.md"
+
+### JSON
+
+!INCLUDE "../common/json-path-notation.md"
+
+### XML
+
+!INCLUDE "../common/xml-path-notation.md"
+
+### DICOM
+
+!INCLUDE "../common/dicom-path-notation.md"
+
+### XLS/XLSX 
+
+!INCLUDE "../common/xls-path-notation.md"
+
+### HL7 
+
+!INCLUDE "../common/hl7-path-notation.md"
+
+### VCF 
+
+Similar as CSV
+
+## Performing consistent masking <a name="persistent"></a>
+
+Consistent masking ensures that all instances of a given original value will be masked to the same fictionalized value. There are two ways to perform consistent masking:
+
+1. Wrap a `MaskingProvider` to a `LocallyPersistentMaskingProvider`. Following the example above:
+
+```java
+MaskingProvider maskingProvider = maskingProviderFactory.get("name");
+MaskingProvider consistentMaskingProvider = new
+					LocallyPersistentMaskingProvider(maskingProvider);
+String maskedValue = consistentMaskingProvider.mask("John Smith");
+```
+
+2. Set `persistence.export` to `true` on the per field configuration. See the masking configuration page for more details
+
```

Comment:
Would it be better to somehow have this part closer to the configurations for consistent masking?